phial-scope/src/main/java/com/mindcoders/phial/internal/PhialScopeNotifier.java

```diff
@@ -26,13 +26,13 @@ static void removeListener(OnScopeChangedListener listener) {
         LISTENERS.remove(listener);
     }
 
-    protected static void fireEnterScope(String scope) {
+    protected static void fireEnterScope(Scope scope) {
         for (OnScopeChangedListener listener : LISTENERS) {
             listener.onEnterScope(scope);
         }
     }
 
-    protected static void fireExitScope(String scope) {
+    protected static void fireExitScope(Scope scope) {
```

Comment:
Scope contains view, so current api requires passing view in order to exit from scope.
Let's keep it simple and pass just name.